firebase-database/src/main/java/com/google/firebase/database/core/SyncTree.java

```diff
@@ -551,60 +591,16 @@ public Node getServerValue(QuerySpec query) {
           @Override
           public List<? extends Event> call() {
             final QuerySpec query = eventRegistration.getQuerySpec();
-            Path path = query.getPath();
-
-            Node serverCacheNode = null;
-            boolean foundAncestorDefaultView = false;
-            // Any covering writes will necessarily be at the root, so really all we need to find is
-            // the server cache. Consider optimizing this once there's a better understanding of
-            // what actual behavior will be.
-            // for (Map.Entry<QuerySpec, View> entry: views.entrySet()) {
-            {
-              ImmutableTree<SyncPoint> tree = syncPointTree;
-              Path currentPath = path;
-              while (!tree.isEmpty()) {
-                SyncPoint currentSyncPoint = tree.getValue();
-                if (currentSyncPoint != null) {
-                  serverCacheNode =
-                      serverCacheNode != null
-                          ? serverCacheNode
-                          : currentSyncPoint.getCompleteServerCache(currentPath);
-                  foundAncestorDefaultView =
-                      foundAncestorDefaultView || currentSyncPoint.hasCompleteView();
-                }
-                ChildKey front =
-                    currentPath.isEmpty() ? ChildKey.fromString("") : currentPath.getFront();
-                tree = tree.getChild(front);
-                currentPath = currentPath.popFront();
-              }
-            }
-
-            SyncPoint syncPoint = syncPointTree.get(path);
-            if (syncPoint == null) {
-              syncPoint = new SyncPoint(persistenceManager);
-              syncPointTree = syncPointTree.set(path, syncPoint);
-            } else {
-              foundAncestorDefaultView = foundAncestorDefaultView || syncPoint.hasCompleteView();
-              serverCacheNode =
-                  serverCacheNode != null
-                      ? serverCacheNode
-                      : syncPoint.getCompleteServerCache(Path.getEmptyPath());
-            }
-
-            persistenceManager.setQueryActive(query);
-
-            CacheNode serverCache;
-            if (serverCacheNode != null) {
-              serverCache =
-                  new CacheNode(IndexedNode.from(serverCacheNode, query.getIndex()), true, false);
-            } else {
+            GetServerCacheNodeResult result = getCacheNodeForQuery(query);
+            CacheNode serverCache = result.node();
+            if (serverCache == null) {
               // Hit persistence
               CacheNode persistentServerCache = persistenceManager.serverCache(query);
               if (persistentServerCache.isFullyInitialized()) {
                 serverCache = persistentServerCache;
               } else {
-                serverCacheNode = EmptyNode.Empty();
-                ImmutableTree<SyncPoint> subtree = syncPointTree.subtree(path);
+                Node serverCacheNode = EmptyNode.Empty();
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.